textbook/chapter_1.qmd

```diff
@@ -0,0 +1,993 @@
+"How is a cars fuel efficiency related to its horsepower?"
+
+::: {.panel-tabset}
+
+## Altair
+
+```{python}
+alt.Chart(cars).mark_point().encode(
+    x='Miles_per_Gallon',
+    y='Horsepower'
+)
+```
+
+## ggplot
+
+```{python}
+%%R
+ggplot(cars, aes(x = Miles_per_Gallon, y = Horsepower)) +
+    geom_point()
+```
+
+::: {.column-margin}
+
+<br>
+<br>
+The default ggplot style is quite different from that of Altair.
+Both have advantages,
+and they are easy to customize to look however you want,
+which we will also get into later.
+Note that ggplot does not including the origin of the plot (x=0, y=0) by default,
+but Altair does for many cases.
+
+:::
+::: 
+
+Here we can see that there seems to be a negative correlation with Horsepower and Miles_per_Gallon:
+More powerful cars consume more gas.
+
+But there might be other variables that also influence this relationship,
+e.g. the weight of the car probably matters for fuel efficiency,
+since more force is needed to propel a bigger mass.
+We can add more dimensions to this chart,
+e.g. specifying which column we want to color the points by
+and Altair will automatically figure out an appropriate colorscale to use.
+
+
+::: {.panel-tabset}
+
+# Altair
+```{python}
+alt.Chart(cars).mark_point().encode(
+    x='Miles_per_Gallon',
+    y='Horsepower',
+    color='Weight_in_lbs'
+)
+```
+
+# ggplot
+
+```{python}
+%%R -w 500 -h 375 
+
+ggplot(cars, aes(x = Miles_per_Gallon, y = Horsepower, color = Weight_in_lbs)) +
+    geom_point()
+    
+```
+
+::: {.column-margin}
+<br>
+<br>
+As with Altair,
+ggplot chooses an appropriate continuous color scale for the quantitative data,
+although it start from dark instead of light,
+which could be related to the different choice of background color.
+Notice also the difference in how the legend is added,
+in Altair the overall width of the figure was increased,
+whereas here the legend takes part of the figure width from the canvas,
+which is made narrower.
+:::
+
+:::
+
+Aha! Here we can see that weight is also correlated with Horsepower (positively)
+and Miles_per_Gallon (negatively).
+We can't say anything about causation from the plot alone,
+just correlation.
+
+One of the really neat things with Altair is that it extends its grammar 
+to interactive plot elements as well.
+We will see much more of this later in the book.
+Let's add a tooltip to each data point with the name of the car,
+hover over the data points to see the tooltip show up!
+
+
+```{python}
+alt.Chart(cars).mark_point().encode(
+    x='Miles_per_Gallon',
+    y='Horsepower',
+    tooltip='Name'
+)
+```
+
+
+## Mission
+
+Your name is **Zephyr Eco-Luna** and you have been tasked with investigating emissions in the city of Viridia. With a commitment to sustainability and a vision for a greener future, Viridia is taking proactive steps to investigate and mitigate its environmental impact. One significant area of focus is the emissions associated with food consumption and production.
```

Comment:
A mission in the end, interesting! I was originally thinking that we will disperse all the exercises throughout, but maybe a combination could work and I remember we were also thinking about having these exercises at the end as a summary so that makes sense.

I really like the creativity and the scenario you are setting up. I think it is of high quality and engaging. I'm wondering if we should leave the avatar and name for the student's role unspecified and just describe who they are in each scenario. Or maybe the student can always be themselves but they are called upon for helping out by characters that care about different aspects of the planet (that could be Minister Reed here). I'm a bit hesitant to define a look and name for the student because they might not feel like they identify with that avatar.

Even if we let the students be themselves, we should make sure we have a diverse set of different avatars and don't stereotype them too much (e.g. mostly beautiful women and strong men). Something to think about going forward (we might even opt to not use humans, but let's leave that for later)

Reply:
Glad you like the mission idea. I totally agree about being too prescriptive with the avatars. My thinking was that the student could play a diverse range of different characters throughout the book. But I agree that as soon as you start designing the appearance of the student's avatar you are on shaky ground. 

I really think the chat interface with avatars works well though so I'd hope to find a solution that allows us to maintain that in some way. One idea is that students could create their avatar with DALLE. Or we have a pre-selected range of avatars. Or, as you say, bring in the aliens! I quite like the idea of having one main character who helps a lot of people within the city (some could be very grateful whilst others could treat you as the new kid on the block - like Minister Reed). We could still use the ID card thing but just to display your new role.  

With regards to beauty, this is definitely something I was aware of. I tried actively counteracting this but the results were even stranger. There's a reason why almost everyone chooses a fairly symmetrical avatar of themselves in games etc.

Reply:
I tried updating this in my latest push to test out what it would look like without an avatar for the reader and the avatar card being on the NPC instead. It needs tweaking, but let me know what you think of the main idea.